src/util/transaction/transactionUtils.ts

```diff
@@ -104,3 +112,22 @@ export function combineAndRegroupSignerTxns(
     txn: newTxnGroup[index]
   }));
 }
+
+/**
+ * Extracts the account address from the first transaction in the provided transaction group.
+ * @param txGroup - The transaction group to extract the account address from
+ * @returns the account address
+ */
+export function extractAccountAddressFromTxGroup(txGroup: SignerTransaction[]): string {
```

Comment:
This function's only responsibility should be encode the account address for the txn it is given. It should be the caller's responsibility to decide which transaction's account address it needs. We can rename the function as well

```suggestion
export function extractSenderAddress(txn: SignerTransaction): string {
```